libtelemetry/src/full/java/com/mapbox/android/telemetry/provider/MapboxTelemetryInitProvider.java

```diff
@@ -35,8 +35,12 @@
   private final ServiceConnection telemetryServiceConnection = new ServiceConnection() {
     @Override
     public void onServiceConnected(ComponentName name, IBinder service) {
-      MapboxTelemetryService.Binder binder = (MapboxTelemetryService.Binder) service;
-      telemetryService = binder.getService();
+      if (service instanceof MapboxTelemetryService.Binder) {
+        MapboxTelemetryService.Binder binder = (MapboxTelemetryService.Binder) service;
+        telemetryService = binder.getService();
+      } else {
+        Log.e(TAG, "Invalid type of MapboxTelemetryService.Binder=" + service);
```

Comment:
do we want to have it as an `error` or maybe enough to have it as a `warning`?

Reply:
Yes, fixed.